packages/astro/types.d.ts

```diff
@@ -4,11 +4,14 @@ import type { Simplify } from './dist/type-utils.js';
 
 /** Any supported HTML or SVG element name, as defined by the HTML specification */
 export type HTMLTag = keyof astroHTML.JSX.DefinedIntrinsicElements;
+
 /** The built-in attributes for any known HTML or SVG element name */
 export type HTMLAttributes<Tag extends HTMLTag> = Omit<
 	astroHTML.JSX.IntrinsicElements[Tag],
-	keyof Omit<AstroBuiltinAttributes, 'class:list'>
->;
+	keyof Omit<AstroBuiltinAttributes, 'class:list'> 
+> & {
+	[key: string]: string | number | boolean | undefined; 
```

Comment:
@Princesseuh Do you prefer `any`?

Reply:
Without necessarily using `any`, allowing also `null` would be great

Reply:
> Without necessarily using `any`, allowing also `null` would be great

Done